src/main/java/com/alexbt/algos/queue/QueueImplWithStacks1.java

```diff
@@ -0,0 +1,29 @@
+package com.alexbt.algos.queue;
+
+import java.util.Stack;
+
+public class QueueImplWithStacks1<T> {
+    private Stack<T> s1 = new Stack<>();
+    private Stack<T> s2 = new Stack<>();
+
+    public void enqueue(T val) {
+        while (!s1.isEmpty()) {
+            s1.push(s2.pop());
+        }
+        s1.push(val);
+    }
+
+    public T dequeue() {
+        while (!s2.isEmpty()) {
+            s2.push(s1.pop());
+        }
+        return s2.pop();
+    }
+
+    public T peek() {
+        while (!s2.isEmpty()) {
```

Comment:
Could check if s1 is empty, then instead of pop/push, the pointers could be swapped.